bot/cogs/assignable_roles_cog.py

```diff
@@ -19,25 +21,139 @@ def __init__(self, bot) -> None:
     async def roles(self, ctx, *, input_role: str = None) -> None:
 
         if input_role is None:
-            await self.send_channel_list(ctx, 'Assignable Roles')
+            await self.send_role_list(ctx, 'Assignable Roles')
             return
         
         try:
             role = await commands.RoleConverter().convert(ctx, input_role)
+            await self.set_role(ctx, role)
+
+        # If RoleConverter failed
         except BadArgument:
-            await self.send_channel_list(ctx, f'@{input_role} not found')
+            # Casefold the roles
+            str_input_role = str(input_role).casefold()
+            role_list = ctx.guild.roles # Return list of roles
+            str_role_list = [str(i).casefold() for i in role_list] # Case-fold to do case insensitive matching
+
+            # Compare input_role to role_list entries for matches
+            role_count, matching_roles = 0, []
```

Comment:
Did a small rework. `role_count` gets `len()` from `matching_roles` now, instead of using a counter. I like having the variable so I don't have to call the `len()` function every time (I imagine this is also technically faster since the `len()` evaluation doesn't need to be performed multiple times)

Reply:
Using a counter means you have to balance two seperate states and that just adds uneeded complexity as well as creates a new avenue for bugs later on. and as for speed there wont be any speed difference as the length of a list is cached as a variable internally, so len() is just checking a variable, specifically the __len__ dunder. So the speed of lookup will be the same, because they are basically the same thing.